strongbox-web-core/src/main/java/org/carlspring/strongbox/web/RepositoryInstanceMethodArgumentResolver.java

```diff
@@ -0,0 +1,126 @@
+package org.carlspring.strongbox.web;
+
+import org.carlspring.strongbox.configuration.Configuration;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.validation.ElementNotFoundException;
+import org.carlspring.strongbox.validation.ServiceUnavailableException;
+
+import javax.inject.Inject;
+import java.util.Map;
+
+import liquibase.util.StringUtils;
+import org.apache.commons.collections.MapUtils;
+import org.springframework.core.MethodParameter;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import org.springframework.web.servlet.HandlerMapping;
+
+/**
+ * @author Pablo Tirado
+ */
+public class RepositoryInstanceMethodArgumentResolver
+        implements HandlerMethodArgumentResolver
+{
+
+    public static final String NOT_FOUND_STORAGE_MESSAGE = "Could not find requested storage %s.";
+    public static final String NOT_FOUND_REPOSITORY_MESSAGE = "Could not find requested repository %s:%s.";
+    public static final String NOT_IN_SERVICE_REPOSITORY_MESSAGE = "Requested repository %s:%s is out of service.";
+
+    @Inject
+    protected ConfigurationManager configurationManager;
+
+    @Override
+    public boolean supportsParameter(final MethodParameter parameter)
+    {
+
+        // Check parameter annotation type
+        if (!parameter.hasParameterAnnotation(RepositoryMapping.class))
+        {
+            return false;
+        }
+        // Check parameter type.
+        return parameter.getParameterType().equals(Repository.class);
+    }
+
+    @Override
+    public Object resolveArgument(final MethodParameter parameter,
+                                  final ModelAndViewContainer modelAndViewContainer,
+                                  final NativeWebRequest nativeWebRequest,
+                                  final WebDataBinderFactory webDataBinderFactory)
+            throws MissingPathVariableException
+    {
+        final String storageId = validateAndGetPathVariable(parameter, nativeWebRequest, "storageId");
```

Comment:
Could you please try to receive the `Repository ` from request attribute ? See `org.carlspring.strongbox.web.RepositoryRequestCondition#expose`. If there isn't then you should use the logic you actually perform here, Thanks

Reply:
there is also `REPOSITORY_NOT_FOUND_REQUEST_ATTRIBUTE` and `STORAGE_NOT_FOUND_REQUEST_ATTRIBUTE`.

Reply:
@ptirador also you should check that `storageId` and `repositoryId` match with that you have in request attributes.